algolia/algoliasearch-sfcc-b2c
# feat: detailed user agent

This PR adds details to the Algolia user agent, which will look like the following:
`Algolia Salesforce B2C vX.X.X; CM: <compatibility_mode>; JobID: <job_id>; StepID: <step_id>`

### Changes
 
- add a `setJobInfo` method on the indexing API Client, so jobs can register themselves once at the beginning of the execution and all subsequent API calls will use this info when getting the service.
- created a `version.json` file to store the version at a centralized place, that all services use. The information is also put in `algoliaData` so it's available in our UI cartridges. This way we will only have to bump the version in `package.json` and `version.json` when releasing

### How to test :test_tube: 

Jobs:
- Run a job and check the received user agent in the Search API Logs (https://dashboard.algolia.com/apps/APP_ID/monitoring/logs)
  - To be sure that jobs are not sharing the same `algoliaIndexingAPI` instance, I've also tried to run 2 in parallel. Both jobID are visible in the logs.

UI:
- Go to the storefront, make a search request and check that the version is correctly set in the request

---
SFCC-148
